src/test/java/de/tum/in/www1/artemis/util/DatabaseUtilService.java

```diff
@@ -636,6 +662,7 @@ public ModelingSubmission addModelingSubmission(ModelingExercise exercise, Model
         return submission;
     }
 
+    @Transactional  // needed for adding values to persistent hash set (submissions)
```

Comment:
can we avoid using transactional here? I removed it in a previous commit. I think you can just save the objects into the repositories if needed.